src/openApi/openApiWrapper.ts

```diff
@@ -37,7 +36,8 @@ export default class OpenApiWrapper {
             request.requestContext.authorizer?.jwt ?? request.headers.Authorization?.split(' ')[1];
           this.userPrincipal =
             request.requestContext.authorizer?.canonicalId ??
-            jwt.decode(this.userToken)?.[this.canonicalIdKey];
+            parseJwtCanonicalId(this.userToken) ??
+            'unknown';
```

Comment:
I wonder if this could ever mix up different users. Perhaps we could do something like UUID here, to be on the safe side?

Reply:
I can change that, this value is not used during caching though. It is mostly to identify the user, e.g. to have something to put in `createdBy` field. Do you still like UUID there more?

Reply:
I see now, `unknown` makes more sense here :)